### PR TITLE
DCS-1239 Conditions fixes

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: server/services/config/conditions/v2/conditions.ts [hunk 1]
```diff
@@ -40,6 +41,16 @@ export const modifyAdditionalConditions = (someConditions: AdditionalConditions)
       appointmentProfessions.map((condition, index) => (index > 0 ? ` ${condition}` : condition))
     )
   }
+
+  if (!isEmpty(conditions?.REPORT_TO)) {
+    const { reportingTime, reportingDaily, ...rest } = conditions?.REPORT_TO
+
+    conditions.REPORT_TO = {
+      reportingTime: reportingTime ? `at ${reportingTime}` : reportingDaily,
```
Comment (andrewrlee) [inline server/services/config/conditions/v2/conditions.ts:49]: prefix "at" or default to daily if not provided - this is used to render the full report

File: server/services/config/conditions/v2/conditions.ts [hunk 2]
```diff
@@ -64,11 +75,10 @@ export const conditions: ConditionMetadata[] = [
   },
   {
     id: 'ATTEND_ALL',
-    text: 'Attend all appointments arranged for you with [INSERT NAME], a [PSYCHIATRIST / PSYCHOLOGIST / MEDICAL PRACTITIONER] and co-operate fully with any care or treatment they recommend.',
+    text: 'Attend all appointments arranged for you with a [PSYCHIATRIST / PSYCHOLOGIST / MEDICAL PRACTITIONER] and co-operate fully with any care or treatment they recommend.',
     user_input: 'attendAll',
     field_position: {
-      appointmentName: 0,
```
Comment (andrewrlee) [inline server/services/config/conditions/v2/conditions.ts:70]: remove name field as no longer required

File: server/services/config/conditions/v2/conditions.ts [hunk 3]
```diff
@@ -181,8 +191,8 @@ export const conditions: ConditionMetadata[] = [
   },
   {
     id: 'COMPLY_REQUIREMENTS',
-    text: 'To comply with any requirements specified by your supervising officer for the purpose of ensuring that you address your [alcohol abuse / drug abuse / sexual behaviour / violent behaviour / gambling / solvent abuse / anger / debt / prolific behaviour/ offending behaviour] problems at the [NAME OF COURSE / CENTRE].',
-    user_input: 'courseOrCentre',
+    text: 'To comply with any requirements specified by your supervising officer for the purpose of ensuring that you address your [alcohol / drug / sexual / violent / gambling / solvent abuse / anger / debt / prolific / offending behaviour] problems at the [NAME OF COURSE / CENTRE].',
+    user_input: 'courseOrCentreV2',
```
Comment (andrewrlee) [inline server/services/config/conditions/v2/conditions.ts:195]: the risks have changed names so needed to version this separately

File: server/services/config/conditions/v2/conditions.ts [hunk 4]
```diff
@@ -383,21 +393,21 @@ export const conditions: ConditionMetadata[] = [
   },
   {
     id: 'PLACE_OF_WORSHIP',
-    text: 'To only attend places of worship which have been previously agreed swith your supervising officer.',
```
Comment (andrewrlee) [inline server/services/config/conditions/v2/conditions.ts:386]: typo

File: server/services/config/conditions/v2/conditions.ts [hunk 5]
```diff
@@ -383,21 +393,21 @@ export const conditions: ConditionMetadata[] = [
...
     field_position: null,
     group_name: GroupName.FREEDOM_OF_MOVEMENT,
     subgroup_name: null,
   },
   {
     id: 'REPORT_TO',
-    text: 'Report to staff at [NAME OF APPROVED PREMISES / POLICE STATION] at [TIME / DAILY], unless otherwise authorised by your supervising officer.  This condition will be reviewed by your supervising officer on a [WEEKLY / MONTHLY / ETC] basis and may be amended or removed if it is felt that the level of risk you present has reduced appropriately.',
-    user_input: 'reportingDetails',
+    text: 'Report to staff at [NAME OF APPROVED PREMISES / POLICE STATION] [at TIME / DAILY], unless otherwise authorised by your supervising officer.  This condition will be reviewed by your supervising officer on a [WEEKLY / MONTHLY / ETC] basis and may be amended or removed if it is felt that the level of risk you present has reduced appropriately.',
+    user_input: 'reportingDetailsV2',
```
Comment (andrewrlee) [inline server/services/config/conditions/v2/conditions.ts:405]: fixed validation reporting and only prefix at if required for time

File: server/services/config/conditions/v2/fieldConfig.ts [hunk 6]
```diff
@@ -281,7 +276,7 @@ export const additionalConditionsV2 = {
         contains: [
           {
             timeframe: {
-              validationMessage: 'Enter the monitoring timefram',
+              validationMessage: 'Enter the monitoring timeframe',
```
Comment (andrewrlee) [inline server/services/config/conditions/v2/fieldConfig.ts:279]: typo

File: server/services/config/conditions/v2/fieldConfig.ts [hunk 7]
```diff
@@ -356,7 +351,7 @@ export const additionalConditionsV2 = {
           },
           {
             appointmentAddressInDrugsSection: {
-              validationMessage: 'Enter appointment name and address',
+              validationMessage: 'Enter appointment address',
```
Comment (andrewrlee) [inline server/services/config/conditions/v2/fieldConfig.ts:354]: name isn't required as part of content and this is confusing

File: server/services/config/conditions/v2/validation.ts [hunk 8]
```diff
@@ -94,12 +93,12 @@ export default joi.object({
...
-    reportingTime: joi.string().allow('').optional(),
-    reportingDaily: joi.when('reportingTime', {
-      is: '',
-      then: joi.string().required(),
-      otherwise: joi.valid('').optional(),
+    reportingTime: joi.when('reportingDaily', {
+      is: joi.exist(),
+      then: joi.valid('').optional(),
+      otherwise: joi.string().required(),
     }),
```
Comment (andrewrlee) [inline server/services/config/conditions/v2/validation.ts:100]: fix validation

will now show message when both fields present, or no fields present

File: server/views/licenceConditions/includes/courseOrCentreV2.pug [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    input(id='debt' type='checkbox' name='abuseAndBehaviours' value='debt' checked=behaviours.includes('debt'))
+    label(for='debt') debt 
+
+  div.multiple-choice
+    input(id='prolific' type='checkbox' name='abuseAndBehaviours' value='prolific' checked=behaviours.includes('prolific'))
+    label(for='prolific') prolific
+
+  div.multiple-choice
+    input(id='offendingBehaviour' type='checkbox' name='abuseAndBehaviours' value='offending behaviour' checked=behaviours.includes('offending behaviour'))
+    label(for='offendingBehaviour') offending behaviour
```
Comment (louisenorris) [inline server/views/licenceConditions/includes/courseOrCentreV2.pug:48]: Think this require a new line and end of file?

**Review by louisenorris — APPROVED** [review]
LGTM

**Pull request merged**